cdk/lib/acquisition-events-api.ts

```diff
@@ -1,23 +1,159 @@
+import { Effect, Policy, PolicyStatement } from "aws-cdk-lib/aws-iam";
+import { Runtime } from "aws-cdk-lib/aws-lambda";
+import { CfnRecordSet } from "aws-cdk-lib/aws-route53";
 import { CfnInclude } from "aws-cdk-lib/cloudformation-include";
 
+export interface AcquisitionEventsApiProps extends GuStackProps {
+  stack: string;
+  stage: string;
+  certificateId: string;
+  domainName: string;
+  hostedZoneName: string;
+  hostedZoneId: string;
+}
 
 export class AcquisitionEventsApi extends GuStack {
-  constructor(scope: App, id: string, props: GuStackProps) {
+  constructor(scope: App, id: string, props: AcquisitionEventsApiProps) {
     super(scope, id, props);
-    const yamlTemplateFilePath = join(
+
+    const app = "acquisition-events-api";
+
+    //  // ---- Existing CFN template ---- //
+    const yamlTemplateFilePath = path.join(
       __dirname,
       "../..",
       "support-lambdas/acquisition-events-api/cfn.yaml"
     );
-    new CfnInclude(this, "YamlTemplate", {
+    const yamlDefinedResources = new CfnInclude(this, "YamlTemplate", {
       templateFile: yamlTemplateFilePath,
       parameters: {
         Stage: props.stage,
       },
     });
+
+    const commonEnvironmentVariables = {
+      App: app,
+      Stack: this.stack,
+      Stage: this.stage,
+    };
+
+    // ---- API-triggered lambda functions ---- //
+    const acquisitionEventsApiLambda = new GuApiLambda(
+      this,
+      "acquisition-events-api-cdk-lambda",
+      {
+        description:
+          "A lambda that Sends in-app acquisitions (subscriptions) to BigQuery",
+        functionName: `${app}-cdk-${this.stage}`,
+        fileName: `${app}.jar`,
+        handler: "com.gu.acquisitionEventsApi.Lambda::handler",
+        runtime: Runtime.JAVA_8,
+        memorySize: 512,
+        timeout: Duration.seconds(300),
+        environment: commonEnvironmentVariables,
+        // Create an alarm
+        monitoringConfiguration: {
+          http5xxAlarm: { tolerated5xxPercentage: 5 },
+          snsTopicName: "conversion-dev",
+        },
+        app: "acquisition-events-api",
+        api: {
+          id: `${app}-${this.stage}`,
+          description: "API Gateway created by CDK",
+        },
+      }
+    );
+
+    // ---- Alarms ---- //
+    const alarmName = (shortDescription: string) =>
+      `ACQUISITION-EVENTS-API-CDK- ${this.stage} ${shortDescription}`;
+
+    const alarmDescription = (description: string) =>
+      `Impact - ${description}. Follow the process in https://docs.google.com/document/d/1_3El3cly9d7u_jPgTcRjLxmdG2e919zCLvmcFCLOYAk/edit`;
+
+    new GuAlarm(this, "ApiGateway4XXAlarmCDK", {
+      app,
+      alarmName: alarmName("API gateway 4XX response"),
+      alarmDescription: alarmDescription(
+        "Acquisition Events API received an invalid request"
+      ),
+      evaluationPeriods: 1,
+      threshold: 1,
+      snsTopicName: "contributions-dev",
+      actionsEnabled: true,
```

Comment:
I suggest we make this conditional, based on the stage/environment, as we don't want to deploy alarms in `CODE`.

Reply:
Eg. `actionsEnabled: isProd()`

Reply:
I deployed alarms to CODe for testing.So now disabling it now as per the suggestion.Instead of using isProd() function,. I have used actionsEnabled: this.stage === "PROD"